_posts/2021-04-28-#17331.md

```diff
@@ -0,0 +1,165 @@
+---
+layout: pr
+date: 2021-04-28
+title: "Use effective values throughout coin selection"
+pr: 17331
+authors: [achow101]
+components: ["wallet"]
+host: Xekyo
+status: upcoming
+commit: 4ac1add
+---
+
+## Notes
+
+- Wallets maintain their balance as a pool of Unspent Transaction Outputs
+  (UTXOs). An
+  [OutputGroup](https://github.com/bitcoin-core-review-club/bitcoin/blob/4ac1adda9914d845aaea5804af4801ffec53c701/src/wallet/coinselection.h#L72)
+  collects all UTXOs that were sent to the same public key.
+
+- _Coin selection_ refers to the process of picking UTXOs from the wallet's UTXO
+  pool to fund a transaction.
+
+- Beyond the primary goal of funding a transaction, the secondary goals of coin
+  selection are:
+
+    - Minimizing short term and long term fees:  
+      We have to balance two needs, we want to pay the smallest fee now to get a
+      timely confirmation, but we have to keep in mind that all of our UTXOs
+      need to be spent in the future. So, we shouldn't over-optimize locally at
+      the detriment of large future costs. E.g. always using largest-first
+      selection minimizes the current cost, but grinds your UTXO pool to dust.
+
+    - Maintaining financial privacy:  
+      There are a number of heuristics that tracking companies employ to cluster
+      payments and addresses. For example, using inputs with the same output
+      script in two transactions indicates that the two transactions involved
+      the same party. Similarly, it is usually assumed that all inputs of a
+      transaction were controlled by the same entity. Sometimes the privacy
+      and economic considerations are opposed. If you want to understand privacy
+      considerations better, check out the [Privacy
+      article](https://en.bitcoin.it/wiki/Privacy) on the Bitcoin wiki.
+
+    - Reliably move the payment(s) to finalization:  
+      Using unconfirmed inputs can make transactions unreliable. When a foreign
+      input is used, the parent transaction may time out or get replaced. Even
+      using self-sent unconfirmed funds may delay the new transaction if the
+      parent transaction has an extensive ancestry, is extraordinarily large, or
+      used a lower feerate than targeted for the child transaction.
+
+- The process of creating a transaction in Bitcoin Core's wallet roughly
+  follows these steps:
+
+    1. `CreateTransaction()` builds the header and recipient outputs, i.e. the
+       parts of the transaction that will remain fixed throughout the coin
+       selection process.
+
+    2. The feerates are established using `GetMinimumFeeRate()` which falls back
```

Comment:
I think you might want to quickly mention which feerates, otherwise it might not be obvious why there are multiple

Reply:
Mentioned some feerates